js_modules/dagster-ui/packages/ui-core/src/runs/StepLogsDialog.tsx

```diff
@@ -113,9 +114,12 @@ export const StepLogsModalContent = ({
   const [logType, setComputeLogType] = useState<LogType>(LogType.structured);
   const [computeLogUrl, setComputeLogUrl] = React.useState<string | null>(null);
 
-  const firstLogForStep = logs.allNodes.find(
+  const flatLogs = useMemo(() => flattenNodeChunks(logs.allNodeChunks), [logs]);
```

Comment:
Could try using `useThrottledMemo` in case the websocket gets spammed very quickly then we can batch multiple updates to lead to just 1 flattenNodeChunks call.

Reply:
```
import React, { useState, useEffect } from 'react';

function withThrottle(WrappedComponent, delay = 1000) {
  return function ThrottledComponent(props) {
    const [throttledProps, setThrottledProps] = useState(props);

    useEffect(() => {
      const handler = setTimeout(() => setThrottledProps(props), delay);
      return () => clearTimeout(handler);
    }, [props, delay]);

    return <WrappedComponent {...throttledProps} />;
  };
}

export default withThrottle;
```


Or maybe we could use something like the above to throttle the number of re-renders for the logs component.

Reply:
There is throttling behavior in `LogsProvider` to guard against this. It looks like it is working as intended, though it's pretty old code and could maybe be revisited at some point.

Reply:
Yeah I added that throttling. Thinking back on it though I think throttling at the component level would be nicer than just throttling the websocket updates.